File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/M. Lucky Numbers/main.dart
import 'dart:io';

bool isLucky(int num) {
  while (num != 0) {
    int digit = num % 10;
    if (digit != 4 && digit != 7) return false;
    num ~/= 10;
  }
  return true;
}

void main() {
  List<String> input = stdin.readLineSync()!.split(" ");
  int a, b;
  bool hasLucky = false;
  a = int.parse(input[0]);
  b = int.parse(input[1]);

  for (a; a <= b; a++) {
    if (isLucky(a)) {
      stdout.write("$a ");
      hasLucky = true;
    }
  }
  if (!hasLucky) print(-1);
}

File: Codeforces/1. ICPC Assiut University Community/Sheet #2 (Loops)/M. Lucky Numbers/main.cs
// https://codeforces.com/group/MWSDmqGsZm/contest/326175/problem/M

using System;
using static System.Console;
using static System.Math;

namespace Codeforces.Assuit.Contest1
{
    class ProblemM
    {
        static void Main(string[] args)
        {
            string[] input = Console.ReadLine().Split(" ");
            uint a, b, i, j;
            bool isLucky = true;
            uint x = uint.Parse(input[0]);
            b = uint.Parse(input[1]);

            for(a=x; a <= b; a++){
                i = a;
                while(i != 0){
                    j = i%10;
                    if(j!=7 && j!=4) {
                        i = 0;
                        break;
                    }
                    if (i/10 == 0){
                        Write($"{a} ");
                        isLucky = false;
                        break;
                    }
                    i = i/10;
                }
            }
            if(isLucky) WriteLine(-1);
         }
    }
}
